src/Castle.Core/DynamicProxy/ModuleScope.cs

```diff
@@ -450,7 +465,11 @@ private AssemblyName GetAssemblyName(bool signStrongName)
 				}
 				assemblyBuilder = (AssemblyBuilder)StrongNamedModule.Assembly;
 				assemblyFileName = StrongNamedModuleName;
+#if NET9_0_OR_GREATER
+				assemblyFilePath = weakModulePath;
+#else
```

Comment:
This alternative assignment may actually work also for .NET 4.6.2+, as `[Weak|Strong]NamedModule.FullyQualifiedName` is possibly equal to `[weak|strong]ModulePath`.